pandas/core/internals.py

```diff
@@ -4443,6 +4444,8 @@ def _lcd_dtype(l):
             return np.dtype('int%s' % (lcd.itemsize * 8 * 2))
         return lcd
 
+    elif have_int and have_float and not have_complex:
```

Comment:
side issue: may want to use ``np.common_type`` here to return lcd of the itemsize for float.